icu4j/main/tests/core/src/com/ibm/icu/dev/test/util/ICUResourceBundleTest.java

```diff
@@ -1179,4 +1174,101 @@ public void TestAlgorithmicParentFallback() {
             assertEquals("Got wrong locale with LOCALE_ROOT", localeRootExpected, localeRootActual);
         }
     }
+
+    @Test
+    public void TestPersonUnits() {
```

Comment:
Would this test actually fail with the current code & data?
I assume that either the current data has "redundant" values to work around these problems, or the current PersonNameFormatter code has workarounds so that it gets complete data.

I am just worried that the problem is masked by workarounds, and the test may not find the problems that are described below.

Would it be better to reconstruct this problem in testdata, independent of what the real locale data looks like? (Could be in a follow-up PR.)